userspace/libsinsp/event.cpp

```diff
@@ -2007,6 +2038,56 @@ const char* sinsp_evt::get_param_as_str(uint32_t id, OUT const char** resolved_s
 							  flags->name);
 			}
 
+			break;
+		}
+	case PT_MODE:
+		{
+			uint32_t val = *(uint32_t *)payload;
+			SET_NUMERIC_FORMAT(prfmt, param_fmt, PRIo32, PRId32, PRIX32);
+			snprintf(&m_paramstr_storage[0],
+					m_paramstr_storage.size(),
+					prfmt, val);
+
+			const struct ppm_name_value *mode = (const struct ppm_name_value *)m_info->params[id].info;
+			const char *separator = "";
+			uint32_t initial_val = val;
+			uint32_t j = 0;
+
+			while(mode != NULL && mode->name != NULL && mode->value != initial_val)
+			{
+				// If mode is 0, then initial_val needs to be 0 for the mode to be resolved
+				if((mode->value == 0 && initial_val == 0) ||
+				   (mode->value != 0 && (val & mode->value) == mode->value && val != 0))
+				{
+					size_t params_len = j + strlen(separator) + strlen(mode->name);
+					if(m_resolved_paramstr_storage.size() < params_len)
+					{
+						m_resolved_paramstr_storage.resize(m_resolved_paramstr_storage.size() + params_len);
```

Comment:
Since params_len is the new size of the resolved string, I think this resize should just be .resize(params_len + 1), with the +1 taking care of the null that snprintf would put at the end of the string.